### DISCUSSION TITLE
Using yt-dlp

### DISCUSSION THREAD

**Discussion by thewh1teagle** [RustPython/RustPython#5338] [Q&A]
I would like to use [`yt-dlp`](https://github.com/yt-dlp/yt-dlp) directly in Rust. Is there a way to use it with `RustPython`?

**Comment by youknowone** [comment]
If you always have system installed python, `pyo3` project is probably the one you are looking for.

If you are going to create a rust create without a CPython dependency, you can try it with RustPython.

Here is a crate I entirely bundled a python package using RustPython. It will be an example crate for your work.
https://github.com/youknowone/aheui-rust/blob/main/rpaheui/src/lib.rs

Disclaimer: I don't guarantee yt-dlp is RustPython compatible.

**Comment by UnknownSuperficialNight** [comment]
Any update on this?
